flang/lib/Optimizer/Transforms/ArrayValueCopy.cpp

```diff
@@ -101,80 +101,136 @@ class ArrayCopyAnalysis {
 };
 } // namespace
 
-// Recursively trace operands to find all array operations relating to the
-// values merged.
-static void populateSets(llvm::SmallVectorImpl<mlir::Operation *> &reach,
-                         llvm::SmallPtrSetImpl<mlir::Value> &visited,
-                         mlir::Value val) {
-  if (!val || visited.contains(val))
-    return;
-  visited.insert(val);
+namespace {
```

Comment:
The github presentation of the diffs in this file is confusing. I'll take a look via the branch...

Reply:
OK. With better tools, I can tell you're turning the recursive function into a class.